src/game/boe.graphutil.cpp

```diff
@@ -319,6 +319,11 @@ void draw_outd_boats(location center) {
 		if(boat.which_town != 200) continue;
 		if(univ.party.in_boat >= 0 && univ.party.boats[univ.party.in_boat] == boat) continue;
 		location loc = local_to_global(boat.loc);
+		if (is_out()) {
```

Comment:
This seems incorrect for two reasons.

1. Redundant `is_out()` check. This function is `draw_outd_boats`, which is called to draw boats if you're outdoors. It checks `is_out()` before calling it.
2. Is this really the only place this logic is needed? I wonder if perhaps it belongs in `local_to_global()`.

If I understand correctly, this is accounting for a case where the party and the boat/horse are in adjacent sectors (presumably both near the edge of the sector). But, outdoor wandering encounters probably need the same logic, don't they? And possibly other things I haven't thought of…

Reply:
Yes, you can suppress the redundant is_out().

**Note** I'm not a fan of these local_to_global/global_to_local methods either, they generate a lot of errors. Then I just tried to fix the code when I detected a problem, without really thinking if there was another solution less problematic.

Reply:
I've applied this without the redundant `is_out()`. A glance at `draw_monsters()` suggests that it's probably handled there somehow too, but it's something to look out for.